microsoft-edge/webview2/concepts/working-with-local-content.md

```diff
@@ -0,0 +1,635 @@
+---
+title: Working with local content in WebView2 apps
+description: Ways to work with local content in WebView2 apps, including navigating to a file URL, navigating to a string, virtual host name mapping, or the WebResourceRequested event.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.technology: webview
+ms.date: 01/10/2023
+---
+# Working with local content in WebView2 apps
+
+In addition to loading remote content, content can also be loaded locally into WebView2.  There are several approaches that can be used to load local content into a WebView2 control, including: 
+* Navigating to a file URL.
+* Navigating to an HTML string.
+* Virtual host name mapping.
+* Handling the `WebResourceRequested` event.
+
+These approaches are described below.
+
+
+<!-- ====================================================================== -->
+## Navigate to a file URL
+
+WebView2 allows navigations to file URLs, to load basic HTML or a PDF.  This is the simplest and most efficient approach to loading local content.  However, it is less flexible than the other approaches.  Like in a web browser, file URLs are limited in some capabilities:
+*  The document origin will be `null` for a file URL.
+*  For each resource, the full path must be specified.
+
+File URLs behave like they do in the browser.  For example, you can't make an `XMLHttpRequest` (XHR) in a file URL, because you're not working in the context of a webpage.  If the WebView2 control is loading local content, still came from a remote source (such as Bing.com), versus a file that came from the local drive.
+
+
+<!-- ------------------------------ -->
+#### Offline (local) vs. online (remote) content
+
+You must specify the full path of the file, for every resource.  For example:
+
+```
+C:\Users\username\Documents\GitHub\Demos\demo-to-do\index.html
+```
+
+For example, if you switch between online (remote) content, such as Bing.com and offline (local) content, you must specify the virtual domain.
+
+In contrast, if you navigate to a file URL, you have to replace it with the full path each time you switch between remote and local content.  When specifying a file URL, the app navigates to a specific path; that's the only way the app can load local content.
```

Comment:
```suggestion
When specifying a file URL, the app navigates to a specific path, not a domain. As a result, 
```

Reply:
When specifying a file URL, the app navigates to a specific path, not a domain.  As a result, you can't use cross-origin resources for navigating to a file URL, because `document.origin` is null.